pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTMethodCallExpression.java

```diff
@@ -4,17 +4,24 @@
 
 package net.sourceforge.pmd.lang.apex.ast;
 
-import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import net.sourceforge.pmd.annotation.InternalApi;
 
+import com.google.summit.ast.Identifier;
 import com.google.summit.ast.expression.CallExpression;
 
 public class ASTMethodCallExpression extends AbstractApexNode.Single<CallExpression> {
+
+    /** The {@link Identifier}s that constitute the {@link #getFullMethodName() full method name}. */
+    private final List<Identifier> components;
+
     @Deprecated
     @InternalApi
-    public ASTMethodCallExpression(CallExpression callExpression) {
+    public ASTMethodCallExpression(CallExpression callExpression, List<Identifier> components) {
```

Comment:
Thoughts on passing additional data to wrapper nodes through the constructor? Previously, all methods could be determined based on only the properties of the internal node or the structure of the AST. Do these additional parameters make the design too dependent on the tree builder design (i.e. if the tree builder is restructured, the implementation here may have to change)?

Reply:
I think it's a fine solution, particularly if the additional data comes from a sibling in the Summit tree structure.

In general, another strategy would be to construct from a single node that is higher in the Summit tree, such that all properties could be computed from its subtree. But if that higher node is too unrelated to a single CallExpression it might make less sense.